src/collection-view.js

```diff
@@ -151,6 +151,43 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
     return this;
   },
 
+  // An efficient rendering used for filtering. Instead of modifying
+  // the whole DOM for the collection view, we are only adding or
+  // removing the related childrenViews.
+  setFilter: function(filter) {
+    if (!filter || this.filter === filter) {
+      return;
+    }
+    this.triggerMethod('before:filter', this);
```

Comment:
these lifecycle events are new, right? I'm not sure this fits in, since the normal render lifecycle doesn't call these. If we do want these, maybe `before:apply:filter` and `apply:filter` would be a bit more explicit?